### PR TITLE
Extend paginate.ts to support more complex pagination object

### PR DESCRIPTION
**Pull request by paulm17**
Closes: https://github.com/blitz-js/blitz/issues/3923

### What are the changes and their implications?

Adds a new pagination object to the return object in paginate.ts.  The pagination object includes the following:
- pageCount
- pageSize
- totalCount
- from
- to

This allows the developer to create a more complex pagination when using usePaginatedQuery.  Such as:

pagination

## Bug Checklist

- [x] Changeset added (run `pnpm changeset` in the root directory)
- [x] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)

## Feature Checklist

- [x] Changeset added (run `pnpm changeset` in the root directory)
- [x] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)
- [x] Documentation added/updated: https://github.com/blitz-js/blitzjs.com/pull/797

### CODE REVIEW

File: packages/blitz/src/paginate.ts [hunk 1]
```diff
@@ -48,11 +48,15 @@ export async function paginate<QueryResult>({
...
   const nextPage = hasMore ? {take, skip: skip + take} : null
+  const pageCount = Math.floor((count + take - 1) / take)
+  const from = skip + 1
+  const to = skip + take
 
   return {
     items,
     nextPage,
     hasMore,
+    pagination: {pageCount: pageCount, pageSize: take, totalCount: count, from, to},
```
Comment (flybayer) [inline packages/blitz/src/paginate.ts:59]:
```suggestion
    pageCount,
    pageSize,
    from,
    to,
```

Comment (dillonraphael) [reply packages/blitz/src/paginate.ts]: Hey @paulm17, seems your latest commit still doesn't address this comment

**Review by flybayer — APPROVED** [review]
looks good, just need to open a PR for docs!

File: packages/blitz/src/paginate.ts (hunk 1 above)
Comment (paulm17) [reply packages/blitz/src/paginate.ts]: Did commit [1788cd3](https://github.com/blitz-js/blitz/pull/3970/commits/1788cd378a92c402423d16d62ecbb42409eeb6ce) do the required change?

**Comment by paulm17** [comment]
Issue edited with PR for docs.

**Review by dillonraphael — APPROVED** [review]

**Pull request merged**